directory-connector/src/com/untangle/app/directory_connector/ActiveDirectoryManagerImpl.java

```diff
@@ -125,24 +129,18 @@ public List<String> getDomains(){
     }
 
     /**
-     * [getAdapter description]
-     * @param  domain [description]
-     * @return        [description]
+     * Retrieves the List of enabled and non-null/non-empty ActiveDirectoryLdapAdapter.
+     * @return List of enabled ActiveDirectoryLdapAdapter
      */
-    public ActiveDirectoryLdapAdapter getAdapter(String domain){
-        for(ActiveDirectoryLdapAdapter adAdapter : this.adAdapters){
-            if(adAdapter == null){
-                continue;
-            }
-            if(!adAdapter.getSettings().getEnabled()){
-                continue;
-            }
-            if(domain != null && !adAdapter.getSettings().getDomain().equals(domain)){
-                continue;
-            }
-            return adAdapter;
-        }
-        return null;
+    public List<ActiveDirectoryLdapAdapter> getAdapters() {
+        return this.adAdapters.stream()
+                .filter(adAdapter -> adAdapter != null
+                        && adAdapter.getSettings().getLDAPHost() != null
```

Comment:
To reduce cognitive complexity, you can use `StringUtils.isNotEmpty(adAdapter.getSettings().getLDAPHost())` for null and empty check.